resources/assets/lib/beatmapset-search-filters.ts

```diff
@@ -40,65 +110,47 @@ export class BeatmapsetSearchFilters implements BeatmapsetSearchParams {
     });
   }
 
-  @computed
-  get displaySort() {
-    return this.selectedValue('sort');
+  getDefault(key: FilterKey) {
+    switch (key) {
+      case 'nsfw':
+        return String(core.userPreferences.get('beatmapset_show_nsfw'));
+      case 'played':
+        return 'any';
+      case 'status':
+        return 'leaderboard';
+      case 'sort':
+        if (this.query?.trim().length ?? 0 > 0) {
+          return 'relevance_desc';
+        } else if (['pending', 'wip', 'graveyard', 'mine'].includes(this.status ?? '')) {
+          return 'updated_desc';
+        } else {
+          return 'ranked_desc';
+        }
+      default:
+        return null;
+    }
   }
 
-  @computed
-  get queryParams() {
-    const values = this.values;
-
-    return BeatmapsetFilter.queryParamsFromFilters(values);
-  }
-
-  @computed
-  get searchSort() {
-    const [field, order] = this.displaySort.split('_');
-    return { field, order };
+  href(key: FilterKey, value: string | null) {
+    return updateQueryString(null, { [keyToChar[key]]: value });
```

Comment:
maybe add check to remove value if it's the default value? "Has Leaderboard" links to `s=leaderboard` but doesn't exist in the url when actually selected